### PR TITLE
Add logs in staking action receipt if necessary

### PR DESCRIPTION
**Pull request by lizhefeng**

### CODE REVIEW

File: action/protocol/staking/handlers.go [hunk 1]
```diff
@@ -62,7 +64,15 @@ func (p *Protocol) handleCreateStake(ctx context.Context, act *action.CreateStak
 		return nil, errors.Wrapf(err, "failed to store account %s", actionCtx.Caller.String())
 	}
 
-	receipt, err := p.settleAction(ctx, sm, gasFee)
+	log := &action.Log{
+		Address:     p.addr.String(),
+		Topics:      nil,
```
Comment (dustinxie) [inline action/protocol/staking/handlers.go]: i am thinking we should also define diff topics for each type of staking/register action?
then from the log we can know what action it is
and data field may take diff meaning for diff action

File: action/protocol/staking/handlers.go [hunk 2]
```diff
@@ -490,6 +509,7 @@ func (p *Protocol) settleAction(
 		ActionHash:      actionCtx.ActionHash,
 		GasConsumed:     actionCtx.IntrinsicGas,
 		ContractAddress: p.addr.String(),
+		Logs:            logs,
```
Comment (dustinxie) [inline action/protocol/staking/handlers.go]: []*Log{logs}?

File: action/protocol/staking/handlers.go (hunk 1 above)
Comment (lizhefeng) [reply action/protocol/staking/handlers.go]: We can add when we need to. So far I think we can leave it blank.

Comment (dustinxie) [reply action/protocol/staking/handlers.go]: and add logs for other actions?

like we discussed, a valid candidate try to update with another's existing name, this should end up with a receipt of log with status = failure, but the block can still be created

File: action/protocol/staking/handlers.go [hunk 3]
```diff
@@ -62,7 +64,15 @@ func (p *Protocol) handleCreateStake(ctx context.Context, act *action.CreateStak
 		return nil, errors.Wrapf(err, "failed to store account %s", actionCtx.Caller.String())
 	}
 
-	receipt, err := p.settleAction(ctx, sm, gasFee)
+	log := &action.Log{
+		Address:     p.addr.String(),
+		Topics:      nil,
+		Data:        byteutil.Uint64ToBytes(bucketIdx),
```
Comment (CoderZhi) [inline action/protocol/staking/handlers.go]: In rewarding protocol, data is struct with type, address, and amount. To be consistent with rewarding protocol, we may need to do the same thing in the staking protocol. However, we won't be able to do indexing or bloomfilter with the complex data structure. thoughts?

**Pull request merged**